scripts/test-secret-scan.sh

```
@@ -27,30 +50,220 @@ gitleaks="$(gitleaks_bin)"
+	esac
+}
+
+# plant_line <rule-id> — a line PROVED to trip that rule against the default
+# rule set, alone in its own directory. A candidate is only SHAPED like the
+# secret; the entropy floor and the rule's stopwords decide the rest, and an
+# assertion that a plant is "still caught" is worthless if the plant never
+# tripped anything. Bounded retries, then a hard failure — never a skip.
+plant_line() {
+	local rule="$1" line attempt
+	printf '[extend]\nuseDefault = true\n' >"$probe/config.toml"
+	for attempt in 1 2 3 4 5 6 7 8; do
```

Shellcheck (0.11.0)

[warning] attempt appears unused. Verify use (or export if used externally).

(SC2034)